src/lib/dma-trace.c

```diff
@@ -422,8 +427,19 @@ void dtrace_event(const char *e, uint32_t length)
 	/* schedule copy now if buffer > 50% full */
 	if (trace_data->enabled &&
 	    buffer->avail >= (DMA_TRACE_LOCAL_SIZE / 2)) {
-		work_reschedule_default(&trace_data->dmat_work,
-		DMA_TRACE_RESCHEDULE_TIME);
+		now = platform_timer_get(platform_timer);
```

Comment:
ok, wouldn't this be simpler if we just scheduled immediately (say 500 microsecs) when buffer is filling up, not sure why we need the calculations here.

Reply:
Yes, I don't know whether it affects playback and make XRUN. I will test with 500us